### PR TITLE
Fixed Topic Ordering

### PR DESCRIPTION
**Pull request by hedgerh**
The promises used when fetching data return in random order, causing the topics to appear on the page in random order.  It's confusing when jQuery appears at the top of the list, and seems like we're suggesting that newbies start with jQuery.

I also added a .gitignore because I ended up w an npm-debug.log somehow.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+npm-debug.log
```
Comment (arthurvr) [inline .gitignore]: this does not deserve here. It's not project-specific so should be in your global gitignore.

File: assets/js/main.js [hunk 2]
```diff
@@ -1,29 +1,38 @@
-
 (function () {
-
 	// add more files here if needed
-        var files = ['javascript.json', 'text-editors.json', 'jquery.json', 'node.json']
+    var files = ['javascript.json', 'node.json', 'jquery.json', 'text-editors.json']
```
Comment (arthurvr) [inline assets/js/main.js]: use a consistent indentation

File: assets/js/main.js [hunk 3]
```diff
@@ -1,29 +1,38 @@
...
-	}
+    function getFile (filename) {
+        return reqwest({
+            url: '/data/' + filename,
+            type: 'json',
+            method: 'get'
+        })
+    }
+
+    function getTargetElement(id) {
```
Comment (arthurvr) [inline assets/js/main.js]: Keep the code style consistent with line 24. Do you want a space before `(` or not.

File: .gitignore (hunk 1 above)
Comment (hedgerh) [inline .gitignore]: good point. will have it fixed in a minute.

File: assets/js/main.js [hunk 4]
```diff
@@ -1,29 +1,38 @@
...
+        // create placeholder divs to enforce topic order
+        var el = getTargetElement(filename.split('.')[0])
+        document.getElementById('content').appendChild(el)
 
-	.forEach(function (filename) {
-		getFile(filename).then(function (data) {
-			render(data)
-		})
-	})
+        getFile(filename).then(function(data) {
```
Comment (arthurvr) [inline assets/js/main.js]: Same thing. Keep the style of the spacing consistent with line 5.

File: assets/js/main.js [hunk 5]
```diff
@@ -1,29 +1,38 @@
...
+            url: '/data/' + filename,
+            type: 'json',
+            method: 'get'
+        })
+    }
+
+    function getTargetElement(id) {
+        var el = document.createElement('div')
+        el.setAttribute('id', 'topic-' + id)
+        return el;
```
Comment (arthurvr) [inline assets/js/main.js]: use semicolons or don't use them at all - not just once upon a time.

File: assets/js/main.js (hunk 2 above)
Comment (hedgerh) [inline assets/js/main.js]: @arthurvr gah.  text editor issue w tabs vs. spaces.  it is lined up technically. will address it.

**Comment by arthurvr** [comment]
Didn't do any testing but this code LGTM.

**Pull request merged**